bin/install-if-deps-outdated.js

```diff
@@ -15,11 +15,34 @@
 const fs = require( 'fs' );
 const path = require( 'path' );
 const spawnSync = require( 'child_process' ).spawnSync;
+//const debug = require( 'debug' )( 'calypso:install' );
+
+const isEmptyObject = obj => ! obj || Object.keys( obj ).length === 0;
+
+const hasDependenciesToInstall = packageDir => {
+	let pkg;
+	try {
+		const packageFile = path.join( packageDir, 'package.json' );
+		if ( ! fs.existsSync( packageFile ) ) {
+			//debug( '[%s] no package.json', packageFile );
+			return false;
+		}
+		pkg = JSON.parse( fs.readFileSync( packageFile, 'utf8' ) );
+	} catch ( e ) {
+		//debug( 'no package file or invalid package file for %s', packageDir, e );
+		return false;
+	}
```

Comment:
did you think about using `require()` here since `node` can natively import `.json` files?

```js
const hasDependenciesToInstall = packageDir => {
	try {
		const pkg = require( `${ packageDir }/package.json` );
		return ! isEmptyObject( Object.assign( {}, pkg.dependencies, pkg.devDependencies ) );
	} catch ( e ) {
		debug( e.message );
		return false;
	}
}
```

Should `catch` for both the file not existing and also for invalid JSON - I validated this with a simple test from the `node` shell on my command line.

Reply:
Yup, gets weird because `require` resolves the path differently. It doesn't use the working directory like `fs` but instead using the require module finding algorithm. We could build up the path relative to the current filename but it seemed more fragile than using `fs` explicitly.

Reply:
🤦‍♂️